src/main/java/com/pretz/dyntreeserver/service/dto/DynTreeMapper.java

```diff
@@ -0,0 +1,22 @@
+package com.pretz.dyntreeserver.service.dto;
+
+import com.pretz.dyntreeserver.service.DynTreeInput;
+import org.springframework.stereotype.Component;
+
+import javax.validation.Valid;
+
+@Component
+public class DynTreeMapper {
+    public DynTreeInput fromDynTreeDTO(@Valid DynTreeDTO dynTreeDTO) {
+        return new DynTreeInput(
```

Comment:
what about input validation? Is it defined somewhere? I see the `@Valid` annotation, however I haven't found any validation related logic.

Or does it just invoke these tags from `DynTreeDTO`, such as `@NotBlank`, `@Min`, etc?

Reply:
Yes, it's handled by framework. You can look at unit tests, where I call Validator explicitly.